census/tf-single/trainer/model.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Implements the vanilla tensorflow model on single node."""
+
+# See https://goo.gl/JZ6hlH for DNN combined
+
+import tensorflow as tf
+
+def random_normal():
+  """Return random normalizer intializer."""
+  return tf.random_normal_initializer(mean=0.0, stddev=0.1)
+
+
+def inference(x, hidden_units=[100,70,50,25], y_units=2):
+  """Create a Feed forward network running on single node
+
+  Args:
+    x: Feature placeholder input
+    hidden_units: Hidden units
+    y_units: Number of classes
+  """
+  x = tf.to_float(x)
+  previous_units = hidden_units[0]
+
+  with tf.variable_scope("layer_0") as scope:
+    weight = tf.get_variable("weight_0",
+                             [x.get_shape()[1], previous_units],
+                             initializer=random_normal())
+
+    bias = tf.get_variable("bias_0",
+                           [previous_units],
+                           initializer=random_normal())
+
+    hidden_layer = tf.add(tf.matmul(x, weight), bias)
+    hidden_layer = tf.nn.relu(hidden_layer)
+
+  for layer_num, units in enumerate(hidden_units[1:]):
```

Comment:
You should roll everything above into this for loop. (And everything below as well). You have a bunch of redundant code here. 
Something like:
```
layer_sizes = x.get_shape[0:1] + hidden_units
layer_shapes = zip(layer_sizes[:1], layer_sizes[1:])
current_layer = x
for num, layer_shape in enumerate(layer_shapes):
   with tf.variable_scope("layer_{}".format(num)):
       weights = tf.get_variable("weights", layer_shape)
       biases = tf.get_variable("biases", layer_shape[0], initializer=tf.zeros_initializer)
  current_layer = tf.relu(tf.matmul(current_layer, weights) + biases)
return current_layer
```

Could be pretty much this whole method.